gulpfile.js

```diff
@@ -0,0 +1,64 @@
+var gulp = require("gulp");
+var mocha = require("gulp-mocha");
+var babel = require("babel/register");
+var browserify = require("browserify");
+var babelify = require("babelify");
+var source = require('vinyl-source-stream');
+var watchify = require("watchify");
+
+var config = {
+  entryFile: './src/index.js',
+  outputDir: './out/',
+  outputFile: 'bundle.js',
+  testFiles: 'test/**/*.js',
+  srcFiles: 'src/**/*.js'
+};
+
+var bundler;
+function getBundler() {
+  if (!bundler) {
+    var options = Object.create(watchify.args);
+    options.debug = true;
+    bundler = watchify(browserify(config.entryFile, options));
+  }
+  return bundler;
+};
+
+function bundle() {
+  return getBundler()
+    .transform(babelify)
+    .bundle()
+    .on('error', function(err) { console.log('Error: ' + err.message); })
+    .pipe(source(config.outputFile))
+    .pipe(gulp.dest(config.outputDir));
+}
+
+gulp.task('build', function(){
+  return bundle();
+});
+
+gulp.task('compile', ['build'] ,function(){
+  process.exit(0);
```

Comment:
I think its because `gulp build` won't terminate the gulp process, so it can be reused by the `gulp watch` task.